client/shared/src/main/scala/org/http4s/client/websocket/WSClient.scala

```diff
@@ -179,21 +214,18 @@ private[http4s] object WSClient {
       override def connectHighLevel(request: WSRequest) =
         for {
           recvCloseFrame <- Resource.eval(Deferred[F, WSFrame.Close])
-          outputOpen <- Resource.eval(Ref[F].of(false))
           conn <- f(request)
         } yield new WSConnectionHighLevel[F] {
           override def send(wsf: WSDataFrame) = conn.send(wsf)
           override def sendMany[G[_]: Foldable, A <: WSDataFrame](wsfs: G[A]): F[Unit] =
             conn.sendMany(wsfs)
-          override def sendClose(reason: String) =
-            conn.send(WSFrame.Close(1000, reason)) *> outputOpen.set(false)
           override def receive: F[Option[WSDataFrame]] = {
             def receiveDataFrame: OptionT[F, WSDataFrame] =
               OptionT(conn.receive).flatMap { wsf =>
                 OptionT.liftF(wsf match {
                   case WSFrame.Ping(data) if respondToPings => conn.send(WSFrame.Pong(data))
                   case wsf: WSFrame.Close =>
-                    recvCloseFrame.complete(wsf) *> outputOpen.get.flatMap(conn.send(wsf).whenA(_))
+                    recvCloseFrame.complete(wsf) *> conn.send(wsf)
```

Comment:
Please check this one, I was confused what the right thing to do here is b/c of https://github.com/http4s/http4s/pull/5974#discussion_r795245627.

Reply:
Actually, I wonder if the `whenA` was also a mistake, related to:
- https://github.com/typelevel/cats/pull/3893.

Reply:
It is safe to remove `outputOpen` now that it is no longer possible to manually send a close frame before the resource handler.

Reply:
Side note: I agree that `ifA` was a suboptimal idea, but what is bad about `whenA`? The issue that "both branches get executed" doesn't apply here AFAICS.

Reply:
I've been wondering about this too. Maybe I'm confused what `whenA` does, isn't `fa.whenA(cond)` syntax sugar for `F.pure(cond).ifA(fa, F.unit)`?

Reply:
No, `cond` is used "directly", such that it is not affected by the `ifA` issue: https://github.com/typelevel/cats/blob/815db7d7f5b60643953850b555bca8b38f6aadf9/core/src/main/scala/cats/Applicative.scala#L192-L193

Reply:
Oh, I see. That make sense. Thanks! :)

Reply:
These combinators are surprisingly trick at times 😆